elide-contrib/elide-dynamic-config-helpers/src/main/java/com/yahoo/elide/contrib/dynamicconfighelpers/validator/DynamicConfigValidator.java

```diff
@@ -0,0 +1,172 @@
+                    false);
+            if (tableConfigs.isEmpty()) {
+                usage();
+                throw new IllegalStateException("No Table Configs found at location: " + absoluteBasePath
+                        + DynamicConfigHelpers.TABLE_CONFIG_PATH);
+            }
+            for (File tableConfig : tableConfigs) {
+                String tableConfigContent = DynamicConfigHelpers.readConfigFile(tableConfig);
+                validateConfigForMissingVariables(tableConfigContent, variables);
+            }
+            try {
+                elideTableConfig = DynamicConfigHelpers.getElideTablePojo(absoluteBasePath, variables);
+            } catch (Exception e) {
+                throw new IllegalStateException("Error while parsing table config at location: " + absoluteBasePath, e);
+            }
+        } else {
+            usage();
+            throw new IllegalStateException("Table Configs Directory doesn't exists at location: " + absoluteBasePath
+                    + DynamicConfigHelpers.TABLE_CONFIG_PATH);
+        }
+        return isTableConfig;
+    }
+
+    private static boolean exists(String filePath) {
+        return new File(filePath).exists();
+    }
+
+    private static void validateConfigForMissingVariables(String config, Map<String, Object> variables) {
+        Pattern regex = Pattern.compile("<%(.*?)%>");
+        Matcher regexMatcher = regex.matcher(config);
+        while (regexMatcher.find()) {
+            String str = regexMatcher.group(1).trim();
+            if (!variables.containsKey(str)) {
+                throw new IllegalStateException(str + " is used as a variable in either table or security config files "
+                        + "but is not defined in variables config file.");
+            }
+        }
+    }
+
+    private static boolean validateSqlInTableConfig(ElideTableConfig elideTableConfig) {
+        for (Table table : elideTableConfig.getTables()) {
+            if (containsDisallowedWords(table.getSql(), SQL_DISALLOWED_WORDS)) {
+                throw new IllegalStateException("SQL provided in table config contain one of these words: "
+                        + Arrays.toString(SQL_DISALLOWED_WORDS));
+            }
+        }
+        return true;
+    }
+
+    private static boolean validateRoleInSecurityConfig(ElideSecurityConfig elideSecurityConfig) {
+        for (String role : elideSecurityConfig.getRoles()) {
+            if (containsDisallowedWords(role, ROLE_NAME_DISALLOWED_WORDS)) {
+                throw new IllegalStateException("ROLE provided in security config contain one of these words: "
+                        + Arrays.toString(ROLE_NAME_DISALLOWED_WORDS));
+            }
+        }
+        return true;
+    }
+
+    private static boolean containsDisallowedWords(String str, String[] keywords) {
+        return Arrays.stream(keywords).parallel().anyMatch(str.toUpperCase(Locale.ENGLISH)::contains);
```

Comment:
Probably faster not to use parallel here.  We are talking about a dozen or so keywords and a small set of tables.

Reply:
removed parallel